server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -234,7 +235,13 @@ private ArrayList<TranslogReader> recoverFromFiles(Checkpoint checkpoint) throws
             // translog was found.
             for (long i = checkpoint.generation; i >= minGenerationToRecoverFrom; i--) {
                 Path committedTranslogFile = location.resolve(getFilename(i));
-                if (Files.exists(committedTranslogFile) == false) {
+                final Checkpoint readerCheckpoint = i == checkpoint.generation
+                    ? checkpoint
+                    : Checkpoint.read(location.resolve(getCommitCheckpointFileName(i)));
```

Comment:
In case both the translog and the checkpoint files are missing, we'll end up with a less descriptive error message thrown by Checkpoint.read() before the channel is opened in openReader(). 

I don't think it is an issue but probably better to have Checkpoint.read() within the try catch block too

Reply:
Not sure how to do this without a refactoring of how things happen here? I still need to read the checkpoint before I can do the translog read? Even if I pull that into the try the mechanics stay the same? I'd have to rip this apart and open the trans log channel before reading the checkpoint wouldn't I?